MapboxCoreNavigation/NavigationService.swift

```diff
@@ -132,13 +160,20 @@ public class MapboxNavigationService: NSObject, NavigationService {
         eventsManager.resetSession()
         eventsManager.start()
     }
-    
-    private func enterTunnel() {
-        simulate(on: router.routeProgress)
+
+    private func resetGPSCountdown() {
+        // Immediately end simulation if it is occuring.
+        if simulatedLocationSource != nil, simulationMode == .onPoorGPS{
```

Comment:
turn the simulationMode check into a guard. There's no reason to reset the timer if we're not on poor GPS mode.